code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm

```diff
@@ -41,13 +67,14 @@
 		if(IS_SKRELL)
 			M.adjustToxLoss(0.5 * removed)
 			return
-		if(IS_TESHARI)
-			..(M, alien, removed*1.2) // Teshari get a bit more nutrition from meat.
-			return
-		if(IS_UNATHI)
-			..(M, alien, removed*2.25) //Unathi get most of their nutrition from meat.
 	..()
 
+/datum/reagent/nutriment/protein/adjust_nutrition(var/mob/living/carbon/M, var/alien, var/removed)
+	switch(alien)
+		if(IS_TESHARI) removed *= 1.25
+		if(IS_UNATHI) removed *= 2.25
```

Comment:
The alien arg is a passed var from the species datum. This is not that snowflake.